pkg/duck/sinks.go

```diff
@@ -17,50 +17,77 @@ limitations under the License.
 package duck
 
 import (
-	"context"
 	"fmt"
 
 	corev1 "k8s.io/api/core/v1"
-	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"k8s.io/client-go/dynamic"
+	"k8s.io/apimachinery/pkg/api/meta"
 
-	duckapis "github.com/knative/pkg/apis"
-	"github.com/knative/pkg/apis/duck"
+	pkgapisduck "github.com/knative/pkg/apis/duck"
 	duckv1alpha1 "github.com/knative/pkg/apis/duck/v1alpha1"
+
+	"github.com/knative/eventing/pkg/reconciler"
+	"github.com/knative/pkg/tracker"
 )
 
-// GetSinkURI retrieves the sink URI from the object referenced by the given
-// ObjectReference.
-func GetSinkURI(ctx context.Context, dynamicClient dynamic.Interface, sink *corev1.ObjectReference, namespace string) (string, error) {
-	if sink == nil {
+// SinkReconciler is a helper for Sources. It triggers
+// reconciliation on creation of the source's sink.
+type SinkReconciler struct {
+	tracker             tracker.Interface
+	sinkInformerFactory pkgapisduck.InformerFactory
+}
+
+// NewSinkReconciler creates and initializes a new SinkReconciler
+func NewSinkReconciler(opt reconciler.Options, callback func(string)) *SinkReconciler {
+	ret := &SinkReconciler{}
+
+	ret.tracker = tracker.New(callback, opt.GetTrackerLease())
+	ret.sinkInformerFactory = &pkgapisduck.CachedInformerFactory{
```

Comment:
This doesn't go through `EnsureTypeMeta`. @mattmoor can you comment on if that is required? Or is this sufficient?

Reply:
From the looks of it, EnsureTypeMeta is appropriate when the GVK is known, which is in general is not the case for sinks.

Reply:
@mattmoor do you think this is good as-is? Or does it need `EnsureTypeMeta()`?